driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -48,16 +48,21 @@ public class TimeoutContext {
     private Timeout computedServerSelectionTimeout;
     private long minRoundTripTimeMS = 0;
 
-    private MaxTimeSupplier maxTimeSupplier = this::calculateMaxTimeMS;
+    @Nullable
+    private MaxTimeSupplier maxTimeSupplier = null;
```

Comment:
Resetting to null to designate a default seemed clearer, and made the default call more obvious in the context of the logic where this was used (which is one place).